domain/offer/internal/types.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package internal
```

Comment:
@SimonRichardson is opposed to `internal` packages within domains. I think just because it isn't a pattern we've adopted elsewhere.

He will probably want you to move them to `state`, unless we get buy-in from the team to follow this pattern